hw/sensor/src/sensor.c

```diff
@@ -1922,6 +1922,59 @@ sensor_set_thresh(char *devname, struct sensor_type_traits *stt)
     return rc;
 }
 
+/**
+ * Unset the thresholds and comparison algo for a sensor
+ *
+ * @param name of the sensor
+ * @param Ptr to sensor threshold
+ *
+ * @return 0 on success, non-zero on failure
+ */
+int
+sensor_unset_thresh(char *devname, struct sensor_type_traits *stt)
+{
+    struct sensor *sensor;
+    struct sensor_type_traits *stt_tmp;
+    int rc;
+
+    if (!stt) {
+        rc = SYS_EINVAL;
+        goto err;
+    }
+
+    sensor = sensor_get_type_traits_byname(devname, &stt_tmp,
+                                           stt->stt_sensor_type);
+    if (!sensor) {
```

Comment:
maybe if (!sensor || !stt_tmp) {}

btw why we care about stt_tmp since this is not used?